<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>registrationDialog</class>
 <widget class="QDialog" name="registrationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>447</width>
    <height>348</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Register Account</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="3,10,1">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Phone</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,3">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Country Code</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="countryCodeEdit">
          <property name="maxLength">
           <number>10</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Phone Number</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="phoneNumberEdit">
          <property name="maxLength">
           <number>30</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="registrationGroup">
     <property name="title">
      <string>Registration Method</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QRadioButton" name="registerRadio">
        <property name="text">
         <string>Register account</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Request registration code as</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="requestSMSButton">
            <property name="toolTip">
             <string>Request a registration code to be send to the phone number as SMS/text</string>
            </property>
            <property name="text">
             <string>SMS</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="requestCallButton">
            <property name="toolTip">
             <string>Request a registration code to be send to the phone number as voice call</string>
            </property>
            <property name="text">
             <string>Call</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Registration code</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="registrationCodeEdit">
            <property name="toolTip">
             <string>Enter the registration code received via SMS or voice call</string>
            </property>
            <property name="maxLength">
             <number>12</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="registerButton">
            <property name="toolTip">
             <string>Register with this registration code</string>
            </property>
            <property name="text">
             <string>Register</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QRadioButton" name="hasPasswordRadio">
        <property name="text">
         <string>Already registered (and have password)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Password (base64)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="passwordEdit">
          <property name="toolTip">
           <string>If you have a WhatsApp password already enter it here in base64 encoding</string>
          </property>
          <property name="maxLength">
           <number>1000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>OK</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
  <zorder>registrationGroup</zorder>
  <zorder>groupBox_2</zorder>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>registrationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>390</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>162</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasPasswordRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>passwordEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>285</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>registerRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>registrationCodeEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>registerRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>requestSMSButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>registerRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>requestCallButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>359</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>registerRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>registerButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
